### DISCUSSION TITLE
The Sunset path: Moving to StreamController

### DISCUSSION THREAD

**Discussion by coolapso** [streamdeck-linux-gui/streamdeck-linux-gui#203] [Announcements]
Hi everyone,

As many of you know, due to the lack of a better alternative, this project was forked from the original [streandeck-ui](https://github.com/timothycrosley/streamdeck-ui) because its main developer disappeared, and dependencies were updated, resulting in the application breaking. Over time, this project has been maintained by the community, and we even considered [rewriting](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/179) it.

For the last few days, a couple of contributors and I have been evaluating and testing a new StreamDeck-related project: [StreamController](https://github.com/StreamController/StreamController). Instead of having two applications doing the same thing in different ways and wasting contributors' resources, we have decided to shift our focus and join forces with the main developer to move towards this new project and grow it together!

This new project was built from the ground up with plugins and a plugin store in mind. It has a much better and cleaner codebase, a modern UI, and a working Flatpak installation. The main developer, @Core447, has been working on this project for quite some time and has been very responsive and friendly towards us, so please be kind to them and have patience. While [StreamController](https://github.com/StreamController/StreamController) is already quite feature-rich, it is still in beta, and there are certainly features missing from streamdeck-linux-gui. Some might not even make it there, or... hey... you can always build a plugin! 😉 There is currently no one-to-one way of migrating from one application to the other, but we are open to suggestions!

### What will happen to streamdeck-linux-gui?

In the meantime, not much will change. The project will still be around, but I will stop accepting new features into it and will only accept relevant bug fixes. Support and issues will start shifting towards [StreamController](https://github.com/StreamController/StreamController).

I know I mentioned I would not start a new project, and I (we) did not. The opportunity just presented itself, and I hope you understand the reasoning behind this change. I truly believe it's for the better, and every streamdeck-linux-gui user will be much better served with it. As it evolves, it can become even better, which it already is compared to what we have now!

Thank you all for being such a great and supportive community. Let's keep the spirit alive in this new home! 😊

**Comment by Core447** [comment]
Thank you! I have started developing an experimental migration tool in the `page-import` branch of [StreamController](https://github.com/StreamController/StreamController). Feel free to check it out and report any problems.

**Reply by smirgol** [reply]
Is that import functionality still accessible somewhere? I'm currently migrating from streamdeck-ui and would like to keep my current setup - more or less. There's a bunch on it and without running streamdeck-ui, which I can't due to errors, I can't recall all the stuff I had in there :)

**Reply by smirgol** [reply]
nevermind, I found the json with the old settings and copied them over by hand

**Comment by Danie10** [comment]
@4s3ti thanks for alerting us to this, and for your embracing the bigger picture to the benefit of all users. I manually migrated myself after being a bit reluctant at first. But I was soon blown away with the features I found on StreamController. Repeat keys on hold works well in my games now, and the plugins with live updates (clock, CPU %, mem graph, speedtest, etc) are very impressive. It is also calling URLS and returning data to display e.g. my solar battery level from my Home Assistant server.

The UI is a little different, and takes a bit to get fully used to, but works very well. All my existing button functions are working well on StreamController so I've certainly not lost any functionality, and in fact have gained a lot.

**Reply by Core447** [reply]
I'm glad that you're enjoying StreamController. Thanks for your feedback!

**Comment by nekename** [comment]
@coolapso @Core447 Have you checked out my project [OpenDeck](https://github.com/ninjadev64/OpenDeck)? It has full* support for Stream Deck SDK plugins and is stable* (whereas StreamController seems to still be in beta).

**Reply by Core447** [reply]
Yes, I did. I'm actually on your Discord server. OpenDeck looks really nice - amazing work! Even though StreamDontroller is still in beta, I and many others use it every day without any problems.

**Reply by Danie10** [reply]
I wouldn't just judge on version numbers though - Gmail was in beta for 10 years. It is very much down to what actual features or functions you need, and which user interfaces work well for you.

**Reply by nekename** [reply]
> Yes, I did. I'm actually on your Discord server. OpenDeck looks really nice - amazing work! Even though StreamDontroller is still in beta, I and many others use it every day without any problems.

Thanks! Yeah, it was the same story with OpenDeck as it was in beta for ages just because every time I felt it was close to stable someone had a niche problem while most others were fine.

Absolutely love the Gtk4 frontend, by the way! I've wanted to do the same but can't unfortunately as the Stream Deck SDK requires a webview of one sort or another.

**Reply by coolapso** [reply]
Actually never seen it, but to be honest I have so much going on right now that I am just happy that stream-deck support on linux is in very good hands! IMHO, Its good to have alternatives, and also being both projects in so different languages don't think will fragment the contributors effort at all.

For what it's worth, my advice for both projects is  to keep up the great job and please keep a healthy relationship between both of you, instead of going on the "unhealthy" competition way!  I think both projects would only benefit from a healthy relationship and learn from each other, there's space for both. :)

borrowing ideas and advice from each other would only make both projects grow more solid. "heey we are having this issue, how did you tackle it?", "yoo! we did it this way, but it didn't work very well if it was today, we would probably try this other  way!", how wonderful would it be, to make this utopia real ;) 

again ... keep up the great work :)

**Reply by Danie10** [reply]
@ninjadev64 I'm testing it but have a runtime error - logged it on your project.

**Reply by nekename** [reply]
Hey @coolapso :wave:

I didn't actually see your comment the first time around, but stumbled across this again and saw it now. Completely agree, of course.

In the spirit of having alternatives, though, do you think you could consider listing my app alongside StreamController in the README? "Stream Deck on Linux" still finds you at streamdeck-ui (and by extension streamdeck-linux-gui) after all this time, and I think that new Linux users would make similar google searches, while the current readme message gives the impression that there aren't alternatives to StreamController.

No worries if not, it's likely they stumble across this thread from the readme anyway, and this would just be a bad replacement for proper promotion of some sort that I should do at some point.

And to Core, I know you've reached out to me on Discord before, but if you do ever want to, feel free to hit me up again.

Thanks :)
